src/analytic_engine/src/sst/parquet/writer.rs

```diff
@@ -405,67 +411,129 @@ impl<'a> RecordBatchGroupWriter<'a> {
     }
 }
 
-struct ObjectStoreMultiUploadAborter<'a> {
-    location: &'a Path,
-    session_id: String,
-    object_store: &'a ObjectStoreRef,
+struct ObjectStoreMultiUpload {
+    multi_upload: MultipartRef,
+    tasks: FuturesUnordered<UploadPart>,
+    completion_task: Option<BoxFuture<'static, std::result::Result<(), Error>>>,
 }
 
-impl<'a> ObjectStoreMultiUploadAborter<'a> {
-    async fn initialize_upload(
-        object_store: &'a ObjectStoreRef,
-        location: &'a Path,
-    ) -> Result<(
-        ObjectStoreMultiUploadAborter<'a>,
-        Box<dyn AsyncWrite + Unpin + Send>,
-    )> {
-        let (session_id, upload_writer) = object_store
+impl<'a> ObjectStoreMultiUpload {
+    async fn new(object_store: &'a ObjectStoreRef, location: &'a Path) -> Result<Self> {
+        let upload_writer = object_store
             .put_multipart(location)
             .await
             .context(Storage)?;
-        let aborter = Self {
-            location,
-            session_id,
-            object_store,
+
+        let multi_upload = Self {
+            multi_upload: Arc::new(Mutex::new(upload_writer)),
+            tasks: FuturesUnordered::new(),
+            completion_task: None,
         };
-        Ok((aborter, upload_writer))
+
+        Ok(multi_upload)
     }
 
-    async fn abort(self) -> Result<()> {
-        self.object_store
-            .abort_multipart(self.location, &self.session_id)
-            .await
-            .context(Storage)
+    pub fn aborter(&self) -> MultipartRef {
+        self.multi_upload.clone()
+    }
+
+    pub fn poll_tasks(
+        mut self: Pin<&mut Self>,
+        cx: &mut std::task::Context<'_>,
+    ) -> std::result::Result<(), object_store::ObjectStoreError> {
+        if self.tasks.is_empty() {
+            return Ok(());
+        }
+        while let Poll::Ready(Some(res)) = self.tasks.poll_next_unpin(cx) {
+            res?;
+        }
+        Ok(())
     }
 }
 
-async fn write_metadata<W>(
-    mut meta_sink: W,
-    parquet_metadata: ParquetMetaData,
-    meta_path: &object_store::Path,
-) -> writer::Result<usize>
-where
-    W: AsyncWrite + Send + Unpin,
-{
-    let buf = encode_sst_meta_data(parquet_metadata).context(EncodePbData)?;
-    let bytes = buf.as_bytes();
-    let bytes_size = bytes.len();
-    meta_sink.write_all(bytes).await.with_context(|| Io {
-        file: meta_path.clone(),
-    })?;
+impl AsyncWrite for ObjectStoreMultiUpload {
+    // TODO: Currently,the data writing is serial, and data may need to be written
```

Comment:
In older version, the write process is parallel?

If it's, we need to implement it as well.